leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/CaConstraintCertificateVerifier.java

```diff
@@ -46,60 +48,51 @@
  *       certificate usage allows both trust anchors and CA certificates,
  *       the certificate might or might not have the basicConstraints
  *       extension present.
- *
+ * </pre>
+ * 
  * For details about Certificate Usage please see:
  * <a href="https://tools.ietf.org/html/rfc6698#section-2.1.1">rfc6698#section-2.1.1</a> - The Certificate Usage Field
  */
-public class CaConstraintCertificateVerifier extends LeshanCertificateVerifierBase {
+public class CaConstraintCertificateVerifier extends BaseCertificateVerifier {
 
-    public CaConstraintCertificateVerifier(Certificate expectedServerCertificate,
-            X509Certificate[] trustedCertificates) {
-        super(expectedServerCertificate, trustedCertificates);
+    private final Certificate caCertificate;
+    private final X509Certificate[] trustedCertificates;
+
+    public CaConstraintCertificateVerifier(Certificate caCertificate, X509Certificate[] trustedCertificates) {
+        Validate.notNull(caCertificate);
+        Validate.notNull(trustedCertificates);
+        Validate.notEmpty(trustedCertificates);
+        this.caCertificate = caCertificate;
+        this.trustedCertificates = trustedCertificates;
```

Comment:
I move attribute in concrete class this way name is more appropriate

Reply:
OK